WordPress/Sniffs/WP/DeprecatedParametersSniff.php

```diff
@@ -0,0 +1,349 @@
+ * detected if the parameter was deprecated before the minimum supported
+ * WP version; a warning otherwise.
+ * By default, it is set to presume that a project will support the current
+ * WP version and up to three releases before.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.12.0
+ */
+class WordPress_Sniffs_WP_DeprecatedParametersSniff extends WordPress_AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 0.12.0
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'wp_deprecated_parameters';
+
+	/**
+	 * Minimum WordPress version.
+	 *
+	 * This variable allows changing the minimum supported WP version used by
+	 * this sniff by setting a property in a custom ruleset xml file.
+	 *
+	 * Example usage:
+	 * <rule ref="WordPress.WP.DeprecatedParameters">
+	 *  <properties>
+	 *   <property name="minimum_supported_version" value="4.5"/>
+	 *  </properties>
+	 * </rule>
+	 *
+	 * @since 0.12.0
+	 *
+	 * @var string WordPress version.
+	 */
+	public $minimum_supported_version = 4.5;
+
+	/**
+	 * Array of function, argument, and default value for deprecated argument.
+	 *
+	 * The functions are ordered alphabetically.
+	 * Last updated for WordPress 4.8.0.
+	 *
+	 * @since 0.12.0
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *    $target_functions = array(
+	 *        (string) Function name. => array(
+	 *            (int) Target parameter position, 1-based. => array(
+	 *                'value'   => (mixed) Expected default value for the
+	 *                              deprecated parameter. Currently the default
+	 *                              values: true, false, null, empty arrays and
+	 *                              both empty and non-empty strings can be
+	 *                              handled correctly by the process_parameters()
+	 *                              method. When an additional default value is
+	 *                              added, the relevant code in the
+	 *                              process_parameters() method will need to be
+	 *                              adjusted.
+	 *                'version' => (int) The WordPress version when deprecated.
+	 *            )
+	 *         )
+	 *    );
+	 */
+	protected $target_functions = array(
+
+		'add_option' => array(
+			3 => array(
+				'value'   => '',
+				'version' => '2.3.0',
+			),
+		),
+		'comments_link' => array(
+			1 => array(
+				'value'   => '',
+				'version' => '0.72',
+			),
+			2 => array(
+				'value'   => '',
+				'version' => '1.3.0',
+			),
+		),
+		'comments_number' => array(
+			4 => array(
+				'value'   => '',
+				'version' => '1.3.0',
+			),
+		),
+		'convert_chars' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '0.71',
+			),
+		),
+		'discover_pingback_server_uri' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '2.7.0',
+			),
+		),
+		'get_delete_post_link' => array(
```

Comment:
If this list is meant to be in alphabetical order, then `get_delete_post_link` should come after `get_category_parents`.